### PR TITLE
feat: Added new proposed links to the sidebar

### LINKED ISSUE #1857: Make left hand sidebar items collapsible and allow for sub-links
## 🚀 Feature Proposal

The left hand sidebar should make the items collapsible and allow for sub links. 

Sidebar should look something like:

```
- Dashboard
- Patient
  - New Patient
  - Patient List  
- Scheduling
   - New Appointment
   - Appointment Schedule
```

When accessing any route `/patient/**`, the Patients item should be highlighted. 
When accessing any route `/appointments/**` the Scheduling item should be highlighted.

Sub items should also be properly marked active. (E.g. when New Patient screen is shown, Patient Item should be highlighted and New Patient item should be highlighted.  

Clicking on any other section should collapse the previously selected section and expand the newly selected one.

## Example

It should operate very similarly to http://demo.hospitalrun.io/

### PR DESCRIPTION
**Pull request by sotous**
Fixes #1857

**Changes proposed in this pull request:**

- Added two sublinks to Patients link
- One to display Patients List, the other one to display Patient Creator Module
- Added two sublinks to Appointment link
- One to display Schedule and the other one to display the Appointment Generator module
- New added links are highlighted when these modules are displaying
- Links at the top navigation bar disappear when the `Sidebar` is available, otherwise, they'll show up.
- Added expansion UI dynamics to the Patients and Appointments link

### CODE REVIEW

**Comment by sotous** [comment]
hi!, @jackcmeyer just uploaded the solution, would appreciate a lot that you guys check it out.

File: src/locales/enUs/translations/scheduling/index.ts [hunk 1]
```diff
@@ -18,18 +16,15 @@ export default {
         emergency: 'Emergency',
         followUp: 'Follow Up',
         routine: 'Routine',
-        walkIn: 'Walk In',
+        walkUp: 'Walk Up',
```
Comment (fox1t) [inline src/locales/enUs/translations/scheduling/index.ts:19]: Why you changed walk in to walk up? We are using walk in to describe walk-in appointments. Can you please revert this?

Comment (sotous) [reply src/locales/enUs/translations/scheduling/index.ts:19]: I am sorry, I actually didn't, it came with the pull I made to master before pushing my local branch. I'll revert that change right away.

Comment (fox1t) [reply src/locales/enUs/translations/scheduling/index.ts:19]: No worries! You don't have to say sorry! :)

**Comment by fox1t** [comment]
Thanks for your contribution!

**Pull request merged**